### DISCUSSION TITLE
A couple "simple" getting started questions

### DISCUSSION THREAD

**Discussion by mikeerickson** [spatie/laravel-query-builder#543] [Q&A]
I am just getting started with this project and have two simple questions which I can't track down the answer (I have scoured the inter web and docs but no luck). Perhaps I am missing something very basic, but docs don't mention these simple query types anywhere.

1. Assuming I want to perform a simple OR query, I want to find all records which `name` contains `Erickson` OR `state` equals `CA`.  I guess in general, how to determine either AND or OR 

```
GET endpoint?filter[name]=Erickson&state=CA
```

2. I would like to limit which fields are to be returned (without having to use the `fields` attribute in query. As it stands, it currently returns all non `$hidden` fields on the model.

Thanks in advance!

**Comment by dominikb** [comment]
Hi @mikeerickson,

> 1. Assuming I want to perform a simple OR query, I want to find all records which name contains Erickson OR state equals CA. I guess in general, how to determine either AND or OR

as for your question about `or-chaining` filters, I think this is currently only possible using some kind of custom filters. Both `FilterExact::class` and `FilterPartial::class` call `$query->where(...)` or `$query->whereRaw()` which default to `and` chaining the conditions.
In my mind, this could be a nice feature but would surely need more discussion on how it should work.

> 2. I would like to limit which fields are to be returned (without having to use the fields attribute in query. As it stands, it currently returns all non $hidden fields on the model.

Limiting the returned fields is not in the scope of this project. (@AlexVanderbist correct me if I am wrong)
I think this could be realized using [Eloquent resources](https://laravel.com/docs/8.x/eloquent-resources) or [PHP fractal](https://github.com/thephpleague/fractal).
